File: deamhome/src/main/java/sosteam/deamhome/domain/account/entity/AccountStatus.java
package sosteam.deamhome.domain.account.entity;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import sosteam.deamhome.global.attribute.Status;
import sosteam.deamhome.global.entity.BaseEntity;

@Document
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AccountStatus extends BaseEntity {
    @NotNull
    @Indexed(unique = true)
    private String userId;

    @NotNull
    @Indexed(unique = true)
    private String snsId;

    @NotNull
    private Status status = Status.LIVE;

    @DBRef(lazy = true)
    @Setter
    private Account account;
}

File: src/main/kotlin/sosteam/deamhome/domain/account/entity/AccountStatus.kt
package sosteam.deamhome.domain.account.entity

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import sosteam.deamhome.global.attribute.SNS
import sosteam.deamhome.global.attribute.Status
import sosteam.deamhome.global.entity.BaseEntity
import java.time.OffsetDateTime

@Table("account_status")
data class AccountStatus(
	@Id
	var id: Long?,
	// unique column
	@Column("user_id")
	val userId: String,
	
	// unique column
	@Column("sns_id")
	var snsId: String? = null,
	
	val sns: SNS,
	
	val email: String,
	
	var status: Status = Status.LIVE,
	
	@Column("account_id")
	var accountId: Long? = null

) : BaseEntity() {
	@Column("deleted_at")
	var deletedAt: OffsetDateTime? = null
}